pkg/cable/wireguard/driver.go

```diff
@@ -111,15 +114,42 @@ func NewDriver(localSubnets []string, localEndpoint types.SubmarinerEndpoint) (c
 		return nil, fmt.Errorf("failed to configure WireGuard device: %v", err)
 	}
 
-	klog.V(log.DEBUG).Infof("Initialized WireGuard %s with publicKey %s", DefaultDeviceName, pub)
+	klog.V(log.DEBUG).Infof("Created WireGuard %s with publicKey %s", DefaultDeviceName, pub)
 	return &w, nil
 }
 
 func (w *wireguard) Init() error {
+	w.mutex.Lock()
+	defer w.mutex.Unlock()
+
+	klog.V(log.DEBUG).Infof("Initializing WireGuard device for cluster %s", w.localEndpoint.Spec.ClusterID)
```

Comment:
With IPsec, we have seen that sometimes the corporate edge routers/firewall configuration blocks the default UDP ports 4500/501. So, we are supporting non-standard ports during subctl join operation. In wireguard, I see that you are using DefaultListenPort (aka 5871). If we encounter such a situation with wireguard, does wireguard support listening/connecting on user-defined ports?

Reply:
WG can use any UDP port for each endpoint

Reply:
So you can probably think of adding support for specifying custom port in future for WG, similar to IPsec.